src/doc/src/reference/specifying-dependencies.md

```diff
@@ -254,30 +301,50 @@ hello_utils = { path = "hello_utils" }
 ```
 
 This tells Cargo that we depend on a crate called `hello_utils` which is found
-in the `hello_utils` folder (relative to the `Cargo.toml` it’s written in).
+in the `hello_utils` folder, relative to the `Cargo.toml` file it’s written in.
+
+The next `cargo build` will automatically build `hello_utils` and
+all of its dependencies.
 
-And that’s it! The next `cargo build` will automatically build `hello_utils` and
-all of its own dependencies, and others can also start using the crate as well.
-However, crates that use dependencies specified with only a path are not
-permitted on [crates.io]. If we wanted to publish our `hello_world` crate, we
+### No local path traversal
+
+The local paths must point to the exact folder with the dependency's `Cargo.toml`.
+Unlike with `git` dependencies, Cargo does not traverse local paths.
+For example, if `cpp` and `cpp_common` are members of a locally cloned `rust-cpp` repo,
+they have to be referred to by the full path:
+
+```toml
+# git key accepts the repo root URL and Cargo traverses the tree to find the crate
+[dependencies]
+cpp        = { git = "https://github.com/mystor/rust-cpp" }
+cpp_common = { git = "https://github.com/mystor/rust-cpp" }
+
+# path key requires the member name to be included in the local path
+[dependencies]
+cpp        = { path = "../rust-cpp/cpp" }
+cpp_common = { path = "../rust-cpp/cpp_common" }
```

Comment:
Could we update these examples change to `rust-lang/regex` as well?
(and ending with `.git` for git repo URL)